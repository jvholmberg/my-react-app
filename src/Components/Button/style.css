.Button {
  display: inline-block;
  border: none;
  color: #fff;
  line-height: 1.5;
  font-weight: 400;
  cursor: pointer;
}


/**
 * Sizes
 */
.Button-Size-Small {
  padding: .25rem .5rem;
  font-size: .875rem;
}
.Button-Size-Medium {
  padding: .375rem .75rem;
  font-size: 1rem;
}
.Button-Size-Large {
  padding: .5rem 1rem;
  font-size: 1.25rem;
}


/**
 * Types
 */

.Button.Primary {
  background-color: #007bff;
}
.Button.Secondary {
  background-color: #6c757d;
}
.Button.Success {
  background-color: #28a745;
}
.Button.Danger {
  background-color: #dc3545;
}
.Button.Warning {
  background-color: #ffc107;
}
.Button.Info {
  background-color: #17a2b8;
}
.Button.Light {
  background-color: #f8f9fa;
  color: #000;
}
.Button.Dark {
  background-color: #343a40;
}
.Button.Link {
  background-color: transparent;
  color: #000;
}

/**
 *  Behavior
 */
.Button.Primary:hover,
.Button.Secondary:hover,
.Button.Success:hover,
.Button.Danger:hover,
.Button.Warning:hover,
.Button.Info:hover,
.Button.Light:hover,
.Button.Dark:hover,
.Button.Link:hover {
  filter: brightness(85%);
}
