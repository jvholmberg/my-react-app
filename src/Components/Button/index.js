import React from 'react';
import { CraftClassName } from '../';

import './style.css';

/**
 * A regular button. It supports multiple color variations and sizes such as following.
 * It's more or less a slim variant of the bootstrap4 variant.
 * Colors: [primary, secondary, success, danger, warning, info, light, dark, link]
 * Sizes: [large, medium, small]
 * @returns {element} <Button>
 */
const Button = ({
  primary, secondary, success, danger, warning, info, light, dark, link,
  large, medium, small, className, children, ...props,
}) => {
  const type = { primary, secondary, success, danger, warning, info, light, dark, link };
  const size = { large, medium, small };

  let cn = CraftClassName(type, 'Button', 'Link');
  cn = CraftClassName(size, cn, 'Medium', 'Button-Size-');
  if (className) cn = `${className} ${cn}`;
  return (<button className={cn} {...props}>{children}</button>);
}
export default Button;
