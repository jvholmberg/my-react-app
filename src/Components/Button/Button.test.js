import React from 'react';

import Button from './';

describe('Button', () => {
  it('Empty matches snapshot', () => {
    expect(shallow(
      <Button />
    )).toMatchSnapshot();
  });
  it('Primary matches snapshot', () => {
    expect(shallow(
      <Button primary />
    )).toMatchSnapshot();
  });
  it('Danger with className matches snapshot', () => {
    expect(shallow(
      <Button className='className' danger />
    )).toMatchSnapshot();
  });
});
