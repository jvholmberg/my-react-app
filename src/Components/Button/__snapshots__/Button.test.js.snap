// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Danger with className matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    className="className"
    danger={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "className Button Danger Button-Size-Medium",
    },
    "ref": null,
    "rendered": null,
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "className Button Danger Button-Size-Medium",
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Button Empty matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "Button Link Button-Size-Medium",
    },
    "ref": null,
    "rendered": null,
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "Button Link Button-Size-Medium",
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Button Primary matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    primary={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "Button Primary Button-Size-Medium",
    },
    "ref": null,
    "rendered": null,
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "Button Primary Button-Size-Medium",
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
